genesyscloud/routing_email_route/genesyscloud_routing_email_route_proxy.go

```diff
@@ -92,73 +92,65 @@ func (p *routingEmailRouteProxy) deleteRoutingEmailRoute(ctx context.Context, do
 	return p.deleteRoutingEmailRouteAttr(ctx, p, domainId, id)
 }
 
-func getAllRoutingEmailRouteByDomainIdFn(ctx context.Context, p *routingEmailRouteProxy, domains *platformclientv2.Inbounddomainentitylisting, name string) (*map[string][]platformclientv2.Inboundroute, *platformclientv2.APIResponse, error) {
+func getAllRoutingEmailRouteByDomainIdFn(_ context.Context, p *routingEmailRouteProxy, domains []platformclientv2.Inbounddomain, pattern string) (*map[string][]platformclientv2.Inboundroute, *platformclientv2.APIResponse, error) {
+	const pageSize = 100
 	var allInboundRoutes = make(map[string][]platformclientv2.Inboundroute)
 	var apiResponse *platformclientv2.APIResponse
-	for _, domain := range *domains.Entities {
-		var allDomainRoutes []platformclientv2.Inboundroute
+	for _, domain := range domains {
+
+		var allDomainRoutes = make([]platformclientv2.Inboundroute, 0)
 
-		for pageNum := 1; pageNum <= *domains.PageCount; pageNum++ {
-			routes, resp, err := p.routingApi.GetRoutingEmailDomainRoutes(*domain.Id, 100, pageNum, name)
+		for pageNum := 1; ; pageNum++ {
+			routes, resp, err := p.routingApi.GetRoutingEmailDomainRoutes(*domain.Id, pageSize, pageNum, pattern)
 			if err != nil {
 				apiResponse = resp
-				return nil, apiResponse, fmt.Errorf("Failed to get routing email route: %s", err)
+				return nil, apiResponse, fmt.Errorf("failed to get routing email route: %s", err.Error())
 			}
 			if routes.Entities == nil || len(*routes.Entities) == 0 {
 				break
 			}
 			allDomainRoutes = append(allDomainRoutes, *routes.Entities...)
-			allInboundRoutes[*domain.Id] = allDomainRoutes
 		}
+		allInboundRoutes[*domain.Id] = allDomainRoutes
```

Comment:
Moving this to here means the value at key `*domain.Id` won't be overwritten on each page of results